src/ArcGISRuntime.Samples.Shared/Managers/SampleManager.cs

```diff
@@ -1,227 +1,160 @@
             }
-            return categories;
+            return samples;
         }
-#endif
 
         /// <summary>
-        /// Gets all samples as a tree.
+        /// Creates a <c>SearchableTreeNode</c> representing the entire 
+        /// collection of samples, organized by category.
         /// </summary>
-        /// <returns>Return all categories, subcategories and samples.</returns>
-        public List<TreeItem> GetSamplesAsTree()
+        /// <remarks>This is public on purpose. Other solutions that 
+        /// consume this project reference it directly.</remarks>
+        /// <param name="allSamples">A list of all samples.</param>
+        /// <returns>A <c>SearchableTreeNode</c> with all samples organized by category.</returns>
+        public static SearchableTreeNode BuildFullTree(IEnumerable<SampleInfo> allSamples)
         {
-            var categories = new List<TreeItem>();
-
-            foreach (var category in _sampleStructureMap.Categories)
-            {
-                var categoryItem = new TreeItem { Name = category.Name };
-
-                foreach (var subCategory in category.SubCategories)
-                {
-                    if (subCategory.ShowGroup)
-                    {
-                        var subCategoryItem = new TreeItem() { Name = subCategory.Name };
-                        categoryItem.Items.Add(subCategoryItem);
-
-                        if (subCategory.Samples != null)
-                            foreach (var sample in subCategory.Samples)
-                                subCategoryItem.Items.Add(sample);
-                    }
-                    else
-                    {
-                        foreach (var sample in subCategory.Samples)
-                            categoryItem.Items.Add(sample);
-                    }
-                }
+            // This code only supports one level of nesting.
+            return new SearchableTreeNode(
+                "All Samples",
+                allSamples.ToLookup(s => s.Category) // put samples into lookup by category
+                .OrderBy(s => s.Key)
+                .Select(BuildTreeForCategory) // create a tree for each category
+                .ToList());
+        }
 
-                categories.Add(categoryItem);
-            }
-            return categories;
+        /// <summary>
+        /// Creates a <c>SearchableTreeNode</c> representing a category of samples.
+        /// </summary>
+        /// <param name="byCategory">A grouping that associates one category title with many samples.</param>
+        /// <returns>A <c>SearchableTreeNode</c> representing a category of samples.</returns>
+        private static SearchableTreeNode BuildTreeForCategory(IGrouping<string, SampleInfo> byCategory)
+        {
+            // This code only supports one level of nesting.
+            return new SearchableTreeNode(
+                name: byCategory.Key,
+                items: byCategory.OrderBy(si => si.SampleName).ToList()
```

Comment:
This is case-sensitive. Did you meant to do `si.SampleName.ToLower()` to match the way `AllSamples` is sorted?